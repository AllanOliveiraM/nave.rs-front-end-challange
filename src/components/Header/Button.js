import React, { Children } from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import Button from 'components/Button'

import { CURRENT_LANGUAGE as lang } from 'helpers/constants'

const ButtonComponent = ({ logout }) => {
  return <StyledButton onClick={logout}>{lang.accounts.logOut}</StyledButton>
}

const StyledButton = styled(Button)`
  background-color: transparent;
  color: black;
  width: auto;
  border: none;
  margin-left: 3.2rem;
`

ButtonComponent.propTypes = {
  logout: PropTypes.func
}

export default ButtonComponent
