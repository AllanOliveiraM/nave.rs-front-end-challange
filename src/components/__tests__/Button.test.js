import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Button from 'components/Button'

describe('<Button />', () => {
  it('render and generate snapshot', () => {
    const component = renderer.create(<Button disabled />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
