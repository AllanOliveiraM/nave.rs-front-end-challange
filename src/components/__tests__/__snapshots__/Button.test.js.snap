// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> render and generate snapshot 1`] = `
.c0 {
  width: regular;
  height: small;
  color: white;
  border-radius: 4px;
}

<button
  className="c0"
  color="white"
  disabled={true}
  height="small"
  width="regular"
/>
`;
